architecture/index.md

```diff
@@ -39,6 +39,55 @@ API resources in the second row (PipelineRun, Snapshot) should primarilly be tho
+  remove, or change [Components] over the lifespan of an [Application].
+- A [Snapshot] represents a collection of particular OCI artifact, mapped to the [Components] of an
+  [Application]. A [Snapshot] is owned by an [Application]. Generally, the [integration-service]
+  produces new [Snapshots] automatically in response to completed builds. It is also possible for
+  the user to construct and provide [Snapshots] to test or release deliberate combinations of
+  OCI artifacts. Old [Snapshots] are garbage collected if not bound to other resources like
+  a [SnapshotEnvironmentBinding] or a [Release].
+- An [IntegrationTestScenario] represents a test that should be run against new [Snapshots] that
+  belong to a particular [Application]. It is owned by an [Application]. The user provides writes
+  tekton pipelines that test their application and registers them with the system by creating
+  [IntegrationTestScenarios].
+- An [Environment] represents a destination that [application-service] and the [gitops-service] can
+  deploy to. Multiple [Applications] in the same workspace can deploy to the same
+  [Environment]. An [Environment] may be long lived (like the `development` [Environment] provided
+  by the system as part of the initialization of a user's workspace) or may be short-lived and
+  ephemeral (like the temporary [Environments] created by the [integration-service] as a part of its
+  test execution processes).
+- A [DeploymentTargetClaim] represents a request to provision a new deployment target. An
+  [Environment] is backed by a [DeploymentTargetClaim]. The [DeploymentTargetClaim] causes new
+  deployment targets to be provisioned which are represented by [DeploymentTargets] that get bound
+  to it. These are typically created and destroyed by the [integration-service] as a part of its
+  test execution processes.
+- A [SnapshotEnvironmentBinding] represents a request to deploy a particular set of OCI artifacts
+  (represented by a [Snapshot]) to a particular location (represented by an [Environment]).
+  A [SnapshotEnvironmentBinding] is owned by an [Application]. The [integration-service promotes
+  OCI artifacts] by updating the [SnapshotEnvironmentBinding] associated with an [Environment]. The
+  [SnapshotEnvironmentBinding] is generally long-lived.
+- A [ReleasePlan] represents a release pipeline that can be used to release a [Snapshot] to some
+  destination, depending on the implementation of the release pipeline. A [ReleasePlan] is owned by
+  an [Application]. It can operate in two modes, one which executes a "tenant" release pipeline in
+  the user's workspace, and another when used in conjunction with a [ReleasePlanAdmission] where it
+  executes a "managed" release pipeline in a separate privileged workspace owned by another team.
+  The [ReleasePlan] is generally long-lived.
+- A [ReleasePlanAdmission] represents an *acceptance* of release pipeline content from another
```

Comment:
Is this used as a gating mechanism when content is being handed off to another workspace, then? Such that the release plan can be resolved and allowed to wait for someone to accept it via the admission object?